File: 프로그래머스/1/12906. 같은 숫자는 싫어/같은 숫자는 싫어.js
function solution(arr)
{
    var answer = [];
    const len=arr.length;
    for(let i=0; i<len;){
        answer.push(arr[i]);
        // 다음 i와 다른 첫 번째 요소를 찾기 위해 현재 인덱스를 기준으로 찾기 시작
        // let Array=[...arr].slice(i+1);
        // let index=Array.findIndex(element => element !== arr[i]);
        let index=i+1;
        while (index < len && arr[index] === arr[i]) {
            index++;
        }
        i = index;
    }
    
    return answer;
}

File: 프로그래머스/1/12906. 같은 숫자는 싫어/같은 숫자는 싫어.java
import java.util.*;

public class Solution {
    public int[] solution(int []arr) {
        List<Integer> list = new ArrayList<>();
        int curr=-1;
        for(int n : arr){
            if(curr!=n){
                curr=n;
                list.add(n);
            }
        }
        int[] answer = list.stream().mapToInt(i -> i).toArray();
        return answer;
    }
}